vlasovsolver/vlasovmover.cpp

```diff
@@ -233,12 +305,44 @@ void calculateSpatialTranslation(
       phiprof::start(profName);
       SpatialCell::setCommunicatedSpecies(popID);
       //      std::cout << "I am at line " << __LINE__ << " of " << __FILE__ << std::endl;
-      calculateSpatialTranslation(mpiGrid,localCells,local_propagated_cells,
-                                  local_target_cells,remoteTargetCellsx,remoteTargetCellsy,
-                                  remoteTargetCellsz,dt,popID);
+      calculateSpatialTranslation(
+         mpiGrid,
+         localCells,
+         local_propagated_cells,
+         local_target_cells,
+         remoteTargetCellsx,
+         remoteTargetCellsy,
+         remoteTargetCellsz,
+         nPencils,
+         dt,
+         popID,
+         time
+      );
       phiprof::stop(profName);
    }
    
+   if (Parameters::prepareForRebalance == true) {
+      if(P::amrMaxSpatialRefLevel == 0) {
+//          const double deltat = (MPI_Wtime() - t1) / local_propagated_cells.size();
+         for (size_t c=0; c<localCells.size(); ++c) {
+//            mpiGrid[localCells[c]]->parameters[CellParams::LBWEIGHTCOUNTER] += time / localCells.size();
+            for (uint popID=0; popID<getObjectWrapper().particleSpecies.size(); ++popID) {
+               mpiGrid[localCells[c]]->parameters[CellParams::LBWEIGHTCOUNTER] += mpiGrid[localCells[c]]->get_number_of_velocity_blocks(popID);
```

Comment:
So let's discuss this. I added the value knowing full well it would swamp the (few tens of) millisecond timings of acceleration. Do we want to make a compound? We're dominated by translation by quite some margin at the moment. Anything based on (time spent in trans) is prone to the non-locality issues of Lb we had been theorising.

Reply:
The acc LBweight addition is commented out now - but if we're able to improve performance with MPI changes in the future, we might want to aggregate these two weights as they might be 50-50.

But you're right in that as long as we have MPI waiting time inside trans, it might end up being a bad metric. So when I get to looking at local ghost translation, I'll do the change then if it works out.

Reply:
So? Keep as is for now?

Reply:
Yep might as well.